File: app/src/main/java/com/kazakago/cleanarchitecture/domain/repository/CityRepository.java
package com.kazakago.cleanarchitecture.domain.repository;

import com.kazakago.cleanarchitecture.domain.model.city.CityModel;

import java.io.IOException;

import io.reactivex.Observable;

/**
 * City Repository
 *
 * @author Kensuke
 */
public interface CityRepository {

    Observable<CityModel> findAll() throws IOException;

}

File: app/src/main/java/com/kazakago/cleanarchitecture/domain/repository/CityRepository.kt
package com.kazakago.cleanarchitecture.domain.repository

import com.kazakago.cleanarchitecture.domain.model.city.CityModel
import io.reactivex.Observable
import java.io.IOException

interface CityRepository {

    @Throws(IOException::class)
    fun findAll(): Observable<CityModel>

}
